Source/CDLCSegment.m

```diff
@@ -37,7 +37,7 @@ - (id)initWithDataCursor:(CDMachOFileDataCursor *)cursor;
         _segmentCommand.cmdsize = [cursor readInt32];
         
         _name = [cursor readStringOfLength:16 encoding:NSASCIIStringEncoding];
-        memcpy(_segmentCommand.segname, [_name UTF8String], sizeof(_segmentCommand.segname));
+        strncpy(_segmentCommand.segname, [_name UTF8String], sizeof(_segmentCommand.segname) - 1);
```

Comment:
Also, do you have to worry about UTF-8 issues? E.g. if the `sizeof(_segmentCommand.segname) - 1`th byte comes in the middle of a UTF-8 byte sequence such that this truncation results in a malformed UTF-8 string?